docs/guides/concepts/1241-docker.md

```diff
@@ -0,0 +1,485 @@
+
+The `entrypoint` field exists only for compatibility reasons. Avoid it if possible. In the end, it just gets prepended to the command to run so you can use `command: name` instead.
+
+:::tip
+
+If you want to ignore the image's entrypoint when running your command, you can clear it by setting it to an empty list:
+
+```cue
+docker.#Run & {
+    entrypoint: []
+    command: ...
+}
+```
+
+:::
+
+#### Command
+
+The `command` field has 3 components: a `name` string, an `args` list and a `flags` struct. They will be combined in the following manner: `<name> <flags> <args>`.
+
+Flag values can either be `true` for just adding the field name, or a `string` to append to the flag.
+
+For example:
+
+```cue
+command: {
+    name: "bash"
+    args: ["/run.sh", "-l", "debug"]
+    flags: {
+        "--norc": true
+        "-e":     true
+        "-u":     true
+        "-o":     "pipefail"
+    }
+}
+
+// will produce
+cmd: ["bash", "--norc", "-e", "-u", "-o", "pipefail", "/run.sh", "-l", "debug"]
+```
+
+#### Secret environment variables
+
+Unlike the image metadata, `docker.#Run` environment variables support `dagger.#Secret` values as well as strings. It's a very simple way to access a secret from a command.
+
+You can read more on this in [Using secrets in a `docker.#Run`](../../core-concepts/1204-secrets.md#in-a-dockerrun).
+
+#### Mounts
+
+The following mount types are available:
+
+- Secret ([example](../../core-concepts/1204-secrets.md#in-a-dockerrun))
+- cache
+- temporary directory
+- directory
+- network socket ([example](../../core-concepts/1203-client.md#using-a-local-socket))
+
+Always specify `contents` and `dest` fields.
+
+The type of the mount is inferred from the value of the `contents` field:
+
+```cue file=../../tests/guides/docker/mounts.cue
+```
+
+<!--FIXME: create separate guide on using mounts and link here-->
+
+#### Exports
+
+It's very common to want to use `core.#ReadFile` to get a file's string contents that a command produced, a `core.#Subdir` to extract a sub directory from the resulting image as a `dagger.#FS` or even `core.#NewSecret` to get the contents of a file as a `dagger.#Secret`. `docker.#Run` allows you to *export* all of that in a very convenient way:
+
+```cue
+_run: docker.#Run & {
+    // mounts, command, etc
+
+    export: {
+        // notice: you can have multiple paths
+        // under each of these fields
+        files: "/output.txt": _
+        secrets: "/token.txt": _
+        directories: "/app/dist": _
+    }
+}
+
+// reference in other fields
+output: _run.export.files."/output.txt"     // string
+token:  _run.export.secrets."/token.txt"    // dagger.#Secret
+dist:   _run.export.directories."/app/dist" // dagger.#FS
+```
+
+:::tip
+
+As in [Use *top* to match anything](../code/1226-coding-style.md#use-top-to-match-anything), the *export* fields `files`, `secrets` and `directories` are already sufficient to declare the type, so we use *top* (`_`) as a simpler alternative to this:
```

Comment:
> As in [Use *top* to match anything](../code/1226-coding-style.md#use-top-to-match-anything), the *export* fields `files`, `secrets` and `directories` are already sufficient to declare the type, **_so in the example above_** we use *top* (`_`) as a simpler alternative to this:

I'd recommend a phrase pointing to the example above so folks look for the hard-to-spot `_` chars at the ends of the lines above.